UATs sometimes failing because of katib tests

### Bug Description

When running the UATs for bundle 1.9, the katib tests often fails with 

```
ERROR    test_notebooks:test_notebooks.py:69 Cell In[9], line 8, in assert_experiment_succeeded(client, experiment)
      1 @retry(
      2     wait=wait_exponential(multiplier=2, min=1, max=10),
      3     stop=stop_after_attempt(30),
      4     reraise=True,
      5 )
      6 def assert_experiment_succeeded(client, experiment):
      7     """Wait for the Katib Experiment to complete successfully."""
----> 8     assert client.is_experiment_succeeded(name=experiment), f"Katib Experiment was not successful."
AssertionError: Katib Experiment was not successful.
```

### To Reproduce

Running UATs against UATs `main` branch result in this failure, see e.g. 

[bundle 1.9](https://github.com/canonical/bundle-kubeflow/actions/runs/13330962426/job/37349598371)
[bundle 1.8](https://github.com/canonical/bundle-kubeflow/actions/runs/13380823243/job/37368996834)

### Environment

UATs branch: `main`
bundle-kubeflow: `main` using 1.9 bundle and using 1.8 bundle, see links above

### Relevant Log Output

```shell
ERROR    test_notebooks:test_notebooks.py:69 Cell In[9], line 8, in assert_experiment_succeeded(client, experiment)
      1 @retry(
      2     wait=wait_exponential(multiplier=2, min=1, max=10),
      3     stop=stop_after_attempt(30),
      4     reraise=True,
      5 )
      6 def assert_experiment_succeeded(client, experiment):
      7     """Wait for the Katib Experiment to complete successfully."""
----> 8     assert client.is_experiment_succeeded(name=experiment), f"Katib Experiment was not successful."
AssertionError: Katib Experiment was not successful.
```

### Additional Context

_No response_

--- a/.github/cluster.yaml
+++ b/.github/cluster.yaml
@@ -27,7 +27,7 @@ managedNodeGroups:
   name: ng-d06bd84e
   releaseVersion: ""
   ssh:
-    allow: false
+    allow: true
   tags:
     alpha.eksctl.io/nodegroup-name: ng-d06bd84e
     alpha.eksctl.io/nodegroup-type: managed

--- a/.github/workflows/deploy-to-eks.yaml
+++ b/.github/workflows/deploy-to-eks.yaml
@@ -110,7 +110,8 @@ jobs:
           kubectl get nodes
 
       - name: Configure EKS Nodes
-        run: |          
+        run: |         
+          echo "Configuring sysctl on EKS workers"
           source ./scripts/gh-actions/set_eks_sysctl_config.sh
 
       - name: Setup juju

--- a/.github/workflows/deploy-to-eks.yaml
+++ b/.github/workflows/deploy-to-eks.yaml
@@ -102,9 +102,48 @@ jobs:
           VERSION_WITHOUT_DOT="${VERSION//.}"
           yq e ".metadata.name |= \"kubeflow-test-$VERSION_WITHOUT_DOT\"" -i .github/cluster.yaml
           yq e ".metadata.version |= \"${{ env.K8S_VERSION }}\"" -i .github/cluster.yaml
+
+          # Generate ssh-key pair
+          ssh-keygen -q -t rsa -N '' -f ~/.ssh/id_rsa <<<y >/dev/null 2>&1
+
           eksctl create cluster -f .github/cluster.yaml
           kubectl get nodes
-      
+
+      - name: Configure EKS Nodes
+        run: |
+          
+          # Export all instances
+          aws ec2 describe-instances \
+            --filters Name=instance-state-name,Values=running \
+            --output yaml > instances.yaml
+
+          INTERNAL_DNS=$(kubectl get nodes -A | tail -n+2 | awk '{print $1}')
+                    
+          for DNS in $INTERNAL_DNS;
+          do
+            # Figure out external dns
+            DNS_QUERY=".Reservations[].Instances | flatten | map(select(.PrivateDnsName==\"${DNS}\"))[].PublicDnsName"
+            echo $DNS_QUERY
+            EXTERNAL_DNS=$(yq "$DNS_QUERY" instances.yaml)
+
+            echo "=========================================="
+            echo "Internal DNS: ${DNS}"
+            echo "Using external DNS: ${EXTERNAL_DNS}"
+            SSH_CMD="ssh -o StrictHostKeyChecking=no ubuntu@$EXTERNAL_DNS"
+          
+            # Setting the sysctl properties
+            echo "Setting values"
+            $SSH_CMD 'sudo sysctl fs.inotify.max_user_watches=655360'
+            $SSH_CMD 'sudo sysctl fs.inotify.max_user_instances=1280'
+          
+            # To check the settings
+            echo "Checking values"
+            $SSH_CMD 'sudo sysctl -a | grep "^fs.inotify"'
+            echo "=========================================="
+          done
+          
+          rm instances.yaml
+
       - name: Setup juju
         run: |
           # Call juju bin directly as a workaround to https://bugs.launchpad.net/juju/+bug/2007575

--- a/.github/workflows/deploy-to-eks.yaml
+++ b/.github/workflows/deploy-to-eks.yaml
@@ -110,39 +110,8 @@ jobs:
           kubectl get nodes
 
       - name: Configure EKS Nodes
-        run: |
-          
-          # Export all instances
-          aws ec2 describe-instances \
-            --filters Name=instance-state-name,Values=running \
-            --output yaml > instances.yaml
-
-          INTERNAL_DNS=$(kubectl get nodes -A | tail -n+2 | awk '{print $1}')
-                    
-          for DNS in $INTERNAL_DNS;
-          do
-            # Figure out external dns
-            DNS_QUERY=".Reservations[].Instances | flatten | map(select(.PrivateDnsName==\"${DNS}\"))[].PublicDnsName"
-            echo $DNS_QUERY
-            EXTERNAL_DNS=$(yq "$DNS_QUERY" instances.yaml)
-
-            echo "=========================================="
-            echo "Internal DNS: ${DNS}"
-            echo "Using external DNS: ${EXTERNAL_DNS}"
-            SSH_CMD="ssh -o StrictHostKeyChecking=no ubuntu@$EXTERNAL_DNS"
-          
-            # Setting the sysctl properties
-            echo "Setting values"
-            $SSH_CMD 'sudo sysctl fs.inotify.max_user_watches=655360'
-            $SSH_CMD 'sudo sysctl fs.inotify.max_user_instances=1280'
-          
-            # To check the settings
-            echo "Checking values"
-            $SSH_CMD 'sudo sysctl -a | grep "^fs.inotify"'
-            echo "=========================================="
-          done
-          
-          rm instances.yaml
+        run: |          
+          source ./scripts/gh-actions/set_eks_sysctl_config.sh
 
       - name: Setup juju
         run: |

--- a/scripts/gh-actions/set_eks_sysctl_config.sh
+++ b/scripts/gh-actions/set_eks_sysctl_config.sh
@@ -1,5 +1,5 @@
 #!/bin/bash
-
+set -xe
 # This script checks that the sysctl are correctly set on an EKS cluster
 
 function check_sysctl(){

--- a/scripts/gh-actions/set_eks_sysctl_config.sh
+++ b/scripts/gh-actions/set_eks_sysctl_config.sh
@@ -11,7 +11,10 @@ function check_sysctl(){
 
     if [ "$CURRENT" != "$VALUE" ]
     then
+      echo "Failing check for ${KEY}: current ${CURRENT} expected: ${VALUE}"
       exit 1
+    else
+      echo "Check for ${KEY} passed successfully"
     fi
 }
 
@@ -32,6 +35,7 @@ do
   echo "=========================================="
   echo "Internal DNS: ${DNS}"
   echo "Using external DNS: ${EXTERNAL_DNS}"
+
   SSH_CMD="ssh -o StrictHostKeyChecking=no ubuntu@$EXTERNAL_DNS"
 
   # Setting the sysctl properties
@@ -41,7 +45,7 @@ do
 
   # To check the settings
   echo "Checking values"
-  check_sysctl "$SSH_CMD" "fs.inotify.max_user_watches" "65536"
+  check_sysctl "$SSH_CMD" "fs.inotify.max_user_watches" "655360"
   check_sysctl "$SSH_CMD" "fs.inotify.max_user_instances" "1280"
   echo "=========================================="
 done

--- a/scripts/gh-actions/set_eks_sysctl_config.sh
+++ b/scripts/gh-actions/set_eks_sysctl_config.sh
@@ -41,8 +41,8 @@ do
 
   # To check the settings
   echo "Checking values"
-  check_sysctl $SSH_CMD "fs.inotify.max_user_watches" "65536"
-  check_sysctl $SSH_CMD "fs.inotify.max_user_instances" "1280"
+  check_sysctl "$SSH_CMD" "fs.inotify.max_user_watches" "65536"
+  check_sysctl "$SSH_CMD" "fs.inotify.max_user_instances" "1280"
   echo "=========================================="
 done
 

--- a/scripts/gh-actions/set_eks_sysctl_config.sh
+++ b/scripts/gh-actions/set_eks_sysctl_config.sh
@@ -1,0 +1,49 @@
+#!/bin/bash
+
+# This script checks that the sysctl are correctly set on an EKS cluster
+
+function check_sysctl(){
+    SSH_CMD=$1
+    KEY=$2
+    VALUE=$3
+
+    CURRENT=$($SSH_CMD 'sudo sysctl -a 2>/dev/null' | grep "^${KEY}" | cut -d "=" -f 2 | xargs)
+
+    if [ "$CURRENT" != "$VALUE" ]
+    then
+      exit 1
+    fi
+}
+
+# Export all instances
+aws ec2 describe-instances \
+  --filters Name=instance-state-name,Values=running \
+  --output yaml > instances.yaml
+
+INTERNAL_DNS=$(kubectl get nodes -A | tail -n+2 | awk '{print $1}')
+
+for DNS in $INTERNAL_DNS;
+do
+  # Figure out external dns
+  DNS_QUERY=".Reservations[].Instances | flatten | map(select(.PrivateDnsName==\"${DNS}\"))[].PublicDnsName"
+  echo $DNS_QUERY
+  EXTERNAL_DNS=$(yq "$DNS_QUERY" instances.yaml)
+
+  echo "=========================================="
+  echo "Internal DNS: ${DNS}"
+  echo "Using external DNS: ${EXTERNAL_DNS}"
+  SSH_CMD="ssh -o StrictHostKeyChecking=no ubuntu@$EXTERNAL_DNS"
+
+  # Setting the sysctl properties
+  echo "Setting values"
+  $SSH_CMD 'sudo sysctl fs.inotify.max_user_watches=655360'
+  $SSH_CMD 'sudo sysctl fs.inotify.max_user_instances=1280'
+
+  # To check the settings
+  echo "Checking values"
+  check_sysctl $SSH_CMD "fs.inotify.max_user_watches" "65536"
+  check_sysctl $SSH_CMD "fs.inotify.max_user_instances" "1280"
+  echo "=========================================="
+done
+
+rm instances.yaml

